### PR TITLE
doc: add curvefs_architecture.md

### PR DESCRIPTION
**Pull request by ilixiaocui**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?): 

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

File: docs/cn/curvefs_architecture.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+- 每个`partition`管理指定范围的 inode，文件系统根目录的 inode 固定为1
+    - dentry 存放在父目录所在的元数据分片上;
+    - 对于文件 /A/B
+        - 首先找到根目录所在的元数据分片，在该元数据分片上查询 dentry(/A)；
+        - 从 dentry(/A) 中获得 /A 的 inodeid，根据 inodeid 获取对应的元数据分片查询 dentry(/A/B)；
+        - 从 dentry(/A/B) 中获得 /A/B 的 inodeid，从而定位到元数据分片，获取 /A/B 的 inode 信息。
+- `partition`由`copyset`管理，`copyset`是raft复制组
+    - `copyset`和`partition`是一对多的关系；
+    - `copyset`和`partition`都是动态创建的，可以弹性扩容。当前的创建策略比较简单：在满足`copyset`的多个副本在不同的`server`的前提下，按照`metaserver`的管理磁盘的剩余容量进行选择；`partition`同样按照`copyset`所在`metaserver`的剩余容量进行选择；
+    - `partition`管理的 inode-tree 和 dentry-tree 全部缓存在内存中，可以快速查询。
```
Comment (aspirer) [inline docs/cn/curvefs_architecture.md:55]: 全缓存这部分逻辑是不是要修改掉的？

File: docs/cn/curvefs_architecture.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+<img src="../images/curvefs_data_s3.png" alt="curvefs_data_s3.png" width="700">
+
+- 在`curve-fuse`端，一个文件的地址空间由多个固定大小的`chunk`组成，每个`chunk`由多个长度不固定的`datacache`组成；
+- `datacache`按照`block`的粒度拆分后上传到 S3 中；
+- 每个数据块在 inode 中由{ chunkid, offset, len, size } 表示，每个数据块在 S3 中的 key 由 { fsid, inodeid, chunkid, blockindex } 表示。根据 inode 中数据位置的记录，可以通过`chunk`和`block`的大小，计算出这部分数据对应的 S3 中数据块的数量以及每个数据块的 key。
+
+## 系统特性
+1. 多存储系统支持。数据既支持存储到公有云，有支持存储到本地存储系统；
+2. 元数据集群和数据集群高可用、高可扩、高可靠；
+3. 支持高速缓存，客户端有内存和磁盘两级缓存加速；
```
Comment (aspirer) [inline docs/cn/curvefs_architecture.md:95]: 支持多级缓存，BS集群可作为持久化缓存层提供。

File: docs/cn/curvefs_architecture.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+对于文件系统中的一个文件，地址空间和 S3 对象的对应关系如下图所示：
+
+<img src="../images/curvefs_data_s3.png" alt="curvefs_data_s3.png" width="700">
+
+- 在`curve-fuse`端，一个文件的地址空间由多个固定大小的`chunk`组成，每个`chunk`由多个长度不固定的`datacache`组成；
+- `datacache`按照`block`的粒度拆分后上传到 S3 中；
+- 每个数据块在 inode 中由{ chunkid, offset, len, size } 表示，每个数据块在 S3 中的 key 由 { fsid, inodeid, chunkid, blockindex } 表示。根据 inode 中数据位置的记录，可以通过`chunk`和`block`的大小，计算出这部分数据对应的 S3 中数据块的数量以及每个数据块的 key。
+
+## 系统特性
+1. 多存储系统支持。数据既支持存储到公有云，有支持存储到本地存储系统；
```
Comment (aspirer) [inline docs/cn/curvefs_architecture.md:93]: 有支持存储->又支持

支持数据在不同存储系统之家自由流转，支持主动/被动数据生命周期管理。

File: docs/cn/curvefs_architecture.md (hunk 1 above)
Comment (ilixiaocui) [reply docs/cn/curvefs_architecture.md:55]: fix

File: docs/cn/curvefs_architecture.md (hunk 3 above)
Comment (ilixiaocui) [reply docs/cn/curvefs_architecture.md:93]: fix

**Pull request merged**